packages/react-instantsearch/src/core/createConnector.js

```diff
@@ -105,7 +105,7 @@ export default function createConnector(connectorDesc) {
         props: this.getProps(nextProps),
       });
 
-      if (isWidget) {
+      if (isWidget && !shallowEqual(this.props, nextProps)) {
```

Comment:
Right before we update the state, do we need to update the state if props did not change?